src/hausdorff.jl

```diff
@@ -82,35 +82,44 @@ See also: [`hausdorff`](@ref)
 const ModifiedHausdorff = GenericHausdorff{MeanReduction, MaxReduction}
 ModifiedHausdorff() = ModifiedHausdorff(MeanReduction(), MaxReduction())
 
-# convert binary image to a point set format
-function img2pset(img::AbstractArray{T}) where T<:Union{Gray{Bool}, Bool}
-    inds = findall(x->x==true, img)
-    [inds[j][i] for i=1:ndims(img), j=1:length(inds)]
+# convert binary image to its distance transform
+const hausdorff_transform(img::AbstractArray{Bool}) = img |> feature_transform |> distance_transform
+
+const hausdorff_transform(img::GenericGrayImage) = hausdorff_transform(of_eltype(Bool, img))
+
+function evaluate_hausdorff(d::GenericHausdorff,
+                            imgA::AbstractArray{Bool},
+                            imgB::AbstractArray{Bool},
+                            dtA::AbstractArray,
+                            dtB::AbstractArray)
+    # trivial cases
+    imgA == imgB && return 0.
+    (isempty(imgA) || isempty(imgB)) && return Inf
```

Comment:
To avoid type instability:

```suggestion
T = result_type(d, dtA, dtB)
imgA == imgB && return zero(T)
(isempty(imgA) || isempty(imgB)) && return convert(T, Inf)
```

I'm not sure if `Distances.result_type` works out of box, so might need to extend it accordingly.